cloud/services/networkinterfaces/networkinterfaces.go

```diff
@@ -40,7 +40,8 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		default:
 			nicConfig := &network.InterfaceIPConfigurationPropertiesFormat{}
 
-			nicConfig.Subnet = &network.Subnet{ID: to.StringPtr(azure.SubnetID(s.Scope.SubscriptionID(), nicSpec.VNetResourceGroup, nicSpec.VNetName, nicSpec.SubnetName))}
+			subnet := &network.Subnet{ID: to.StringPtr(azure.SubnetID(s.Scope.SubscriptionID(), nicSpec.VNetResourceGroup, nicSpec.VNetName, nicSpec.SubnetName))}
+			nicConfig.Subnet = subnet
```

Comment:
nit: Perhaps, move the assignment and initialization to one multi-line statement. The initialization of the `Subnet` struct is rather long.